File: src/main/java/pd/discordipc/models/ActivityType.java
package pd.discordipc.models;

public enum ActivityType {
    PLAYING(0),
    STREAMING(1),
    LISTENING(2),
    WATCHING(3),
    CUSTOM(4),
    COMPETING(5);

    private final int id;

    ActivityType(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }
}

File: src/main/kotlin/com/pdlayer/discordipc/models/ActivityType.kt
package com.pdlayer.discordipc.models

enum class ActivityType(val id: Int) {
    PLAYING(0),
    STREAMING(1),
    LISTENING(2),
    WATCHING(3),
    CUSTOM(4),
    COMPETING(5);

    val supportedBySetActivity: Boolean
        get() = this in SUPPORTED_BY_SET_ACTIVITY

    private companion object {
        val SUPPORTED_BY_SET_ACTIVITY = setOf(PLAYING, LISTENING, WATCHING, COMPETING)
    }
}
